### ISSUE TITLE
C headers (.h) give out c++ warnings

### ISSUE THREAD

**Issue by MisterChief53** [clangd/clangd#861]
Hello, sorry if this is my mistake due to a lack of understanding of how clangd works. I am editing a c header file (.h), but clangd detects it as a c++ header file and gives out warnings like the following:

```
ISO C++11 does not allow conversion from string literal to 'char *'
```

which was triggered when working with the following line:

```
static char *font = "monospace:pixelsize=12:antialias=true:autohint=true";
```

This results in having so many errors that clangd just stops.
I have a .clangd file in the same directory of the file I am working with, which contains this:

```
If:
  PathMatch: .*.h
CompileFlags:
  Add: [-xc-header]
```

I might have some errors in my code, and the line itself might be wrong even in C (even though it works), but I want to know why is clangd giving me C++ warnings.
I have also tried using `-xc` and `--language=c`, but I still have the same problem.

Clangd is working properly with .c files by the way.

**System information**

Output of `clangd --version`: clangd version 12.0.1

Editor/LSP plugin: coc.nvim, whith coc-clangd

Operating system: Arch Linux x64

**Comment by HighCommander4** [comment]
I believe this is due to a clangd bug which has been fixed in trunk (targeting 13.0): https://github.com/clangd/clangd/issues/555.

Does your project have a compilation database (`compile_commands.json`)? That should allow you to avoid this problem even in earlier versions.

**Comment by MisterChief53** [comment]
> I believe this is due to a clangd bug which has been fixed in trunk (targeting 13.0): #555.
> 
> Does your project have a compilation database (`compile_commands.json`)? That should allow you to avoid this problem even in earlier versions.

Hello, I have generated a compilation database and now everything works as intended. Sorry for the trouble.
